src/test/java/org/cactoos/scalar/AvgOfTest.java

```diff
@@ -51,213 +51,313 @@ public void withEmptyCollection() {
+    @Test
+    public void withDoubleCollectionDoubleValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0d, 2.0d, 3.0d, 4.0d).toArray(new Double[4])
+                1.0d, 2.0d, 3.0d, 4.0d
             ).doubleValue(),
             Matchers.equalTo(2.5d)
         );
+    }
+
+    @Test
+    public void withDoubleCollectionMaxValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0d, 2.0d, 3.0d, 4.0d).toArray(new Double[4])
-            ).floatValue(),
-            Matchers.equalTo(2.5f)
+                Double.MAX_VALUE, Double.MAX_VALUE
+            ).doubleValue(),
+            Matchers.equalTo(Double.MAX_VALUE)
         );
     }
 
     @Test
-    public void withFloatCollection() {
+    public void withDoubleCollectionMinValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0f, 2.0f, 3.0f, 4.0f).toArray(new Float[4])
-            ).intValue(),
-            Matchers.equalTo(2)
+                Double.MIN_VALUE, Double.MIN_VALUE
+            ).doubleValue(),
+            Matchers.equalTo(Double.MIN_VALUE)
         );
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void withDoubleCollectionPositiveInfinity() {
+        new AvgOf(
+            Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY
+        ).doubleValue();
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void withDoubleCollectionNegativeInfinity() {
+        new AvgOf(
+            Double.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY
+        ).doubleValue();
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void withDoubleCollectionNaN() {
+        new AvgOf(
+            Double.NaN, Double.NaN
+        ).doubleValue();
+    }
+
+    @Test
+    public void withDoubleCollectionNegativeNumbersDoubleValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0f, 2.0f, 3.0f, 4.0f).toArray(new Float[4])
-            ).longValue(),
-            Matchers.equalTo(2L)
+                -1.0d, -2.0d, -3.0d, -4.0d
+            ).doubleValue(),
+            Matchers.equalTo(-2.5d)
         );
+    }
+
+    @Test
+    public void withDecimalCollectionPrecisionProblem() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0f, 2.0f, 3.0f, 4.0f).toArray(new Float[4])
-            ).doubleValue(),
-            Matchers.equalTo(2.5d)
+                100.0, 100.666, 100.0
+            ).floatValue(),
+            Matchers.equalTo(100.222f)
         );
+    }
+
+    @Test
+    public void withDecimalCollectionPrecisionProblemExtraDecimalRange() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0f, 2.0f, 3.0f, 4.0f).toArray(new Float[4])
+                100.266, 100.267
             ).floatValue(),
-            Matchers.equalTo(2.5f)
+            Matchers.equalTo(100.2665f)
```

Comment:
@allanws These values are double and you're checking if the result is float. Is is right?

Reply:
you can down/up cast numbers, this is "usual"...